src/client/src/sync/mod.rs

```diff
@@ -135,6 +137,32 @@ mod context {
         }
     }
 
+    impl<S> StoreContext<S>
+    where
+        S: Spec + Send + Sync + 'static,
+        <S as Spec>::Status: Send + Sync,
+        S::IndexKey: Send + Sync,
+    {
+        pub fn watch<'a>(
+            &'a self,
+        ) -> impl Stream<Item = MetadataChanges<S, AlwaysNewContext>> + 'a {
+            let mut listener = self.store.change_listener();
+            let (sender, receiver) = async_channel::unbounded();
+
+            fluvio_future::task::spawn_local(async move {
+                loop {
+                    listener.listen().await;
+                    let changes = listener.sync_changes().await;
+                    if let Err(e) = sender.send(changes).await {
+                        error!("Failed to send Metadata update: {:?}", e);
+                    }
+                }
+            });
```

Comment:
I think this is ok for now.  need to way to terminate if watcher drops

Reply:
You mean to terminate the dispatcher if the watch stream is dropped? Or terminate the stream if something is dropped?

Reply:
Actually, I think this might be ok.  Check behavior of sender  if receiver is dropped.  It would be good if this is not multi sender and receiver

Reply:
So `sender.send` is a future that returns a Result. So if all receivers are dropped, it just returns an `Err`, which I think is fine for our uses.